src/schoology-extractor/data_extractor/lib/sections.py

```diff
@@ -0,0 +1,22 @@
+from .request_client import RequestClient
+
+
+class Sections:
+    """
+    The Sections class contains methods to get sections' information from Schoology API
+
+    Args:
+        SCHOOLOGY_KEY (str): The consumer key given by Schoology
+        SCHOOLOGY_SECRET (str): The consumer secret given by Schoology
+
+    Attributes:
+        request_client (RequestClient): The client used for authenticate and send requests to Schoology
+
+    """
+
+    def __init__(self, SCHOOLOGY_KEY, SCHOOLOGY_SECRET):
+        self.request_client = RequestClient(SCHOOLOGY_KEY, SCHOOLOGY_SECRET)
+
+    def get_by_id(self, section_id):
+        response = self.request_client.get(f"sections/{section_id}")
+        return response
```

Comment:
The various classes like `Sections`, `Submissions`, `Users`, etc. are very small. What do you think about moving `get_by_id` into the `RequestClient` and changing the name to `get_sections`? Then you wouldn't need to instantiate so many objects, turning `RequestClient` into a fatter façade for making calls to the Schoology API.

Reply:
And what do you think about keeping classes but only as properties in RequestClient? I am thinking about keeping files small, if we wrap them inside of RequestClient while keeping the classes for separating the code, we can avoid creating too many instances in the main.py by moving them to the __init__ method of the RequestClient, but we'll keep the possibility of adding more code to the classes without making the RequestClient a single big class.

Reply:
I don't have a _strong_ feeling about this. I've normally used, and normally seen, a single client with lots of methods for getting what you need from a remote server - whether that is an API or a database. Thinking about a database repository class that works on the entire database, or the Entity Framework DbContext class. It is true that I've also seen one repository per table, but it just seems like overkill and more maintenance that is needed.

If keeping separate classes for resource that will be retrieved, then will they benefit from a base class to reduce code duplication?

Reply:
I don't think they could get a lot of benefit from a base class, I tried that and I ended up adding more lines than the ones I save, and the only thinkg I could share was the initial request client, because there are not many reusable methods, because for example a get_all method, will have a different path structure:
users : /users
assignments: sections/{section_id}/assignments

I think we could  go with a single RequestClient for now, if it gets too big, should not be so difficult to separate some methods in the future.